extensions/security/iprange/src/org/exist/security/realm/iprange/IPRangeRealm.java

```diff
@@ -0,0 +1,154 @@
+
+    public IPRangeRealm(final SecurityManagerImpl sm, final Configuration config) throws ConfigurationException {
+        super(sm, config);
+        instance = this;
+    }
+
+    private static long ipToLong(final InetAddress ip) {
+        final byte[] octets = ip.getAddress();
+        long result = 0;
+        for (final byte octet : octets) {
+            result <<= 8;
+            result |= octet & 0xff;
+        }
+        return result;
+    }
+
+	/*@Override
+    public void start(DBBroker broker) throws EXistException {
+		super.start(broker);
+	}*/
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public boolean deleteAccount(final Account account) throws PermissionDeniedException, EXistException, ConfigurationException {
+        // Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    public boolean deleteGroup(final Group group) throws PermissionDeniedException, EXistException, ConfigurationException {
+        // Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    public Subject authenticate(final String ipAddress, final Object credentials) throws AuthenticationException {
+
+        // Elevaste to system privileges
+        try (final DBBroker broker = BrokerPool.getInstance().get(Optional.of(getSecurityManager().getSystemSubject()))) {
+
+            // Convert IP address
+            final long ipToTest = ipToLong(InetAddress.getByName(ipAddress));
+
+            // Get xquery service
+            final XQuery queryService = broker.getBrokerPool().getXQueryService();
+            if (queryService == null) {
+                LOG.error("IPRange broker unable to retrieve XQueryService");
+                return null;
+            }
+
+            // Construct XQuery
+            final String query = "collection('/db/system/security/IPRange/accounts')/account/" +
+                    "iprange[" + ipToTest + " ge number(start) and " + ipToTest + " le number(end)]/../name";
+            final XQueryContext context = new XQueryContext(broker.getBrokerPool());
+
+            final CompiledXQuery compiled = queryService.compile(broker, context, query);
+
+            final Properties outputProperties = new Properties();
+
+            // Execute xQuery
+            final Sequence result = queryService.execute(broker, compiled, null, outputProperties);
+            final SequenceIterator i = result.iterate();
+
+            // Get FIRST username when present
+            final String username = i.hasNext() ? i.nextItem().getStringValue() : "";
+
+            if (i.hasNext()) {
+                LOG.warn("IP address " + ipAddress + " matched multiple ipranges. Using first result only.");
+            }
+
+            if (!"".equals(username)) {
+                final Account account = getSecurityManager().getAccount(username);
+                if (account != null) {
+                    LOG.info("IPRangeRealm trying " + account.getName());
```

Comment:
I think this (and a few of the below statements) could be `LOG.trace` otherwise there is going to be a lot of logging.